lib/spack/spack/build_systems/python.py

```diff
@@ -237,7 +239,12 @@ def test_imports(self) -> None:
                 purpose=f"checking import of {module}",
                 work_dir="spack-test",
             ):
-                python("-c", f"import {module}")
+                setup_context = spack.build_environment.SetupContext(
+                    self.spec, context=spack.context.Context.RUN
+                )
+                mods = setup_context.get_env_modifications()
+                with mods.set_env():
```

Comment:
no need for `set_env`, you can pass the env when calling python(...)

Reply:
Thanks. Commit made but untested for now. Changes to environment.py unneeded now.